hsmd/hsmd.c

```diff
@@ -146,28 +205,39 @@ static void node_key(struct privkey *node_privkey, struct pubkey *node_id)
 					     node_privkey->secret.data));
 }
 
-/**
- * hsm_channel_secret_base -- Derive the base secret seed for per-channel seeds
- *
- * This secret is the basis for all per-channel secrets: the per-channel seeds
- * will be generated mixing in the channel_id and the peer node_id.
- */
+/*~ This secret is the basis for all per-channel secrets: the per-channel seeds
```

Comment:
Comments and code belong together.  Always.

We need to decide where doxygen comments are used: I like them for API headers: not because you would ever extract them, but because you can use them to mark a definitive reference style (as ccan does).

Even better would be a parser which made sure they actually matched the code, though (eg. are all parameter names documented?).